gnucash/import-export/import-account-matcher.c

```diff
@@ -101,29 +101,49 @@ static AccountPickerDialog* gnc_import_new_account_picker(void)
 static gpointer test_acct_online_id_match(Account *acct, gpointer data)
 {
     AccountOnlineMatch *match = (AccountOnlineMatch*)data;
-    const char *current_online_id = gnc_import_get_acc_online_id(acct);
-    int curr_len, match_len;
+    const char *acct_online_id = gnc_import_get_acc_online_id(acct);
+    int acct_len, match_len;
 
-    if (current_online_id == NULL || match->online_id == NULL)
+    if (acct_online_id == NULL || match->online_id == NULL)
         return NULL;
 
-    curr_len = strlen(current_online_id);
+    acct_len = strlen(acct_online_id);
     match_len = strlen(match->online_id);
 
-    if (current_online_id[curr_len - 1] == ' ')
-        --curr_len;
+    if (acct_online_id[acct_len - 1] == ' ')
+        --acct_len;
     if (match->online_id[match_len - 1] == ' ')
         --match_len;
 
-    if (strncmp (current_online_id, match->online_id, curr_len) == 0)
+    if (strncmp (acct_online_id, match->online_id, acct_len) == 0)
     {
-        if (strncmp(current_online_id, match->online_id, match_len) == 0)
+        if (strncmp(acct_online_id, match->online_id, match_len) == 0)
             return (gpointer *) acct;
         if (match->partial_match == NULL)
+        {
             match->partial_match = acct;
-        ++match->count;
+            ++match->count;
+        }
+        else
+        {
+            const char *partial_online_id =
+                gnc_import_get_acc_online_id(match->partial_match);
+            int partial_len = strlen(partial_online_id);
+            if (partial_online_id[partial_len - 1] == ' ')
+                --partial_len;
+            /* Both partial_online_id and acct_online_id are substrings of
+             * match->partial_id, but whichever is longer is the better match.
```

Comment:
I think this comment should refer to match->online_id

Reply:
Right.

Reply:
Just a cosmetic remark:
The pattern 
```c
n = strlen(s);
if (s[n-1] == ' ')
    --n;
```
occurs 3 times within this code. It might be replaced by a function:
```c
size_t effstrlen(const char *s)
{
    size_t len = strlen(s);
    while(len > 0 && s[len - 1] == ' ')
        --len;
    return len;
}
...
n = effstrlen(s);
```
This example would be prepared also for more than one trailing blank.